rust/template/src/main.rs

```diff
@@ -244,31 +252,223 @@ fn main() -> Result<(), String> {
         opt init_snapshot:bool=true, desc:"Do not dump initial output snapshot.";                                                   // --no-init-snapshot
         opt print:bool=true, desc:"Backwards compatibility. The value of this flag is ignored.";                                    // --no-print
         opt workers:usize=1, short:'w', desc:"The number of worker threads. Default is 1.";                                         // --workers or -w
+        opt idle_merge_effort:Option<isize>, desc:"Set Differential Dataflow's 'idle_merge_effort' parameter.This flag takes precedence over the '$DIFFERENTIAL_EAGER_MERGE' environment variable.";
+        opt profile_timely:bool=false, desc:"Use external Timely Dataflow profiler.";
+        opt profile_differential:bool=false, desc:"Use external Differential Dataflow profiler. Implies '--profile-timely'";
+        opt self_profiler:bool, desc:"Enable DDlog internal profiler. This option is mutually exclusive with '--profile-timely'.";
+        opt timely_profiler_socket:Option<String>, desc:"Socket address to send Timely Dataflow profiling events. Default (if '--profile-timely' is specified) is '127.0.0.1:51317'. Implies '--profile-timely'.";
+        opt timely_trace_dir:Option<String>, desc:"Path to a directory to store Timely Dataflow profiling events, e.g., './timely_trace'. Implies '--profile-timely'.";
+        opt differential_profiler_socket:Option<String>, desc:"Socket address to send Differential Dataflow profiling events. Default (if '--profile-differential' is specified is '127.0.0.1:51318'. Implies '--profile-differential'.";
```

Comment:
So I guess you cannot use both a file and a socket

Reply:
Nope, the ddshow client-side library doesn't support that.

Reply:
It does/could, you'd just need to make a tee'd writer a la
```rust
struct Tee<W1, W2> {
    first: W1,
    second: W2,
}

impl<W1, W2> Write for Tee<W1, W2>
where
    W1: Write,
    W2: Write,
{ ... }

ddshow_sink::enable_timely_logging(Tee::new(File::open(..), TcpStream::connect(..)));
```